### ISSUE TITLE
Can't store data in session

### ISSUE THREAD

**Issue by kostyazgara** [SerayaEryn/fastify-session#40]
Hi!
I can't save my data to session. Here is my code
```javascript
const fastify = require('fastify');`
const fastifySession = require('fastify-session');
const fastifyCookie = require('fastify-cookie');

const app = fastify();
app.register(fastifyCookie);
app.register(fastifySession, {
  secret: 'a secret with minimum length of 32 characters',
  cookie: {
    secure: false,
  },
});

app.addHook('preHandler', (request, reply, next) => {
  console.log(request.session.views);

  if (request.session.views) {
    request.session.views++;
  } else {
    request.session.views = 1;
  }

  next();
});

app.get('/', async (req, res) => {
  return {
    views: req.session.views,
  }
});

(async () => {
  await app.listen(3000);
})();
```

I did copy this code from example, but when I try creating new value it doesn't save in session. Before every request, session are reset and my variable undefined

![image](https://user-images.githubusercontent.com/18326076/48909361-baf4f480-ee75-11e8-9cbc-cbf2a8b5547a.png)

I testing in my browser Google Chrome. Session id in cookie are correct and doesn't changes

![image](https://user-images.githubusercontent.com/18326076/48909511-2343d600-ee76-11e8-96a9-5eeefbdf02fe.png)

### My Environment

- Node.js - v10.12.0

- npm - v6.4.1

- fastify - v1.13.0

- fastify-cookie - v2.1.1

- fastify-session - v1.2.0

**Comment by toonvanstrijp** [comment]
I'm having the same problem as well. The session isn't restored correctly.

**Comment by unordinarity** [comment]
I have same problem with same build, can confirm. Also, I use `redis` and `connect-redis` to store data and time-by-time (like randomly) I see valid records in Redis.

**Comment by SerayaEryn** [comment]
The problem is fixed in `v1.2.1`.
